providers/flagd/src/main/java/dev/openfeature/contrib/providers/flagd/grpc/GrpcConnector.java

```diff
@@ -102,43 +105,54 @@ public void shutdown() throws Exception {
         }
     }
 
-    /**
-     * Resets internal state for retrying establishing the stream.
-     */
-    public void resetRetryConnection() {
-        this.eventStreamAttempt = 1;
-        this.eventStreamRetryBackoff = this.startEventStreamRetryBackoff;
-    }
-
     /**
      * Provide the object that can be used to resolve Feature Flag values.
+     *
      * @return a {@link ServiceGrpc.ServiceBlockingStub} for running FF resolution.
      */
     public ServiceGrpc.ServiceBlockingStub getResolver() {
         return serviceBlockingStub.withDeadlineAfter(this.deadline, TimeUnit.MILLISECONDS);
     }
 
-    private void startEventStream() {
-        StreamObserver<Schema.EventStreamResponse> responseObserver =
-                new EventStreamObserver(this.cache, this.stateConsumer, this::restartEventStream);
-        this.serviceStub
-                .eventStream(Schema.EventStreamRequest.getDefaultInstance(), responseObserver);
-    }
+    private void observeEventStream() {
+        while (this.eventStreamAttempt <= this.maxEventStreamRetries) {
+            StreamObserver<Schema.EventStreamResponse> responseObserver =
+                    new EventStreamObserver(sync, this.cache, this::grpcStateConsumer);
+            this.serviceStub.eventStream(Schema.EventStreamRequest.getDefaultInstance(), responseObserver);
+
+            try {
+                synchronized (sync) {
+                    sync.wait();
+                }
+            } catch (InterruptedException e) {
+                log.warn("interruption while waiting for condition", e);
+                return;
+            }
+
+            this.eventStreamAttempt++;
+            this.eventStreamRetryBackoff = 2 * this.eventStreamRetryBackoff;
 
-    private void restartEventStream() {
-        this.eventStreamAttempt++;
-        if (this.eventStreamAttempt > this.maxEventStreamRetries) {
-            log.error("failed to connect to event stream, exhausted retries");
-            this.stateConsumer.accept(ProviderState.ERROR);
-            return;
+            try {
+                Thread.sleep(this.eventStreamRetryBackoff);
+            } catch (InterruptedException e) {
+                log.debug("failed to block while restarting gRPC Event Stream");
```

Comment:
so we don't confuse this with the wait call reading the logs :) 
```suggestion
                log.debug("failed to sleep while restarting gRPC Event Stream");
```